deploy/runtools/run_farm_deploy_managers.py

```diff
@@ -633,6 +640,25 @@ def clear_fpgas(self) -> None:
             for card_bdf in card_bdfs:
                 run(f"xbutil reset -d {card_bdf} --force")
 
+    def localize_xclbin(self, slotno: int) -> None:
+        """ download xclbin URI to remote node. """
+        assert slotno < len(self.parent_node.sim_slots), f"{slotno} can not index into sim_slots {len(self.parent_node.sim_slots)} on {self.parent_node.host}"
+        serv = self.parent_node.sim_slots[slotno]
+        hwcfg = serv.get_resolved_server_hardware_config()
+        assert hwcfg.xclbin is not None
+        if re.match(_RFC_3986_PATTERN, hwcfg.xclbin):
+            remote_home_dir = self.parent_node.get_sim_dir()
+            remote_sim_dir = f"{remote_home_dir}/sim_slot_{slotno}/"
+            hwcfg.local_xclbin = './local.xclbin'
+
+            with cd(remote_sim_dir):
+                run(downloadURI, hwcfg.xclbin, hwcfg.local_xclbin)
```

Comment:
From my testing, this line doesn't compile. run() only accepts a string, which is a shell command, not a function. The 2nd and 3rd arguments are booleans, not arguments to the first function. Am I on the wrong version of fabric or something?  https://github.com/fabric/fabric/blob/1.14/fabric/operations.py#L970

Reply:
I think that we meant `execute()` instead of `run()`.